subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractSmokeTest.groovy

```diff
@@ -175,6 +175,38 @@ abstract class AbstractSmokeTest extends Specification {
         settingsFile = new File(testProjectDir.root, "settings.gradle")
     }
 
+    def withPluginValidation() {
+        def isKotlin = defaultBuildFileName.endsWith('.gradle.kts')
+        def pluginsBlockIdx = buildFile.text.indexOf('plugins {')
+        def buildScriptBlockIdx = buildFile.text.indexOf('buildscript {')
+        if (pluginsBlockIdx > 0 || (pluginsBlockIdx == -1 && buildScriptBlockIdx == -1)) {
+            buildFile.text = buildFile.text.replace('plugins {', """
+            plugins {
+                ${isKotlin ? '`validate-external-gradle-plugin`' : 'id "validate-external-gradle-plugin"'}
+        """)
+        } else if (buildScriptBlockIdx > 0) {
+            throw new UnsupportedOperationException("Validation with buildscript block isn't supported in smoke tests yet")
```

Comment:
"yet": do we plan to support it? What do we lose without it?

Reply:
I think the code code be simplified here if you only checked for `buildScriptBlockIdx > 0` and then went on with the rest of the code outside the `if` block.

Reply:
I don't think we lose anything without it. It's a sanity check, in case a smoke test uses the buildscript block, for some reason, and that it also wants to enable validation. In this case we'll know that we need to do something about it.